In [1]:
import pandas as pd
import requests
import urllib.parse
from tqdm import tqdm

In [2]:
import sys
!{sys.executable} -m pip install --upgrade --force-reinstall geocoder

DEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621
  Using cached geocoder-1.38.1-py2.py3-none-any.whl (98 kB)
  Using cached ratelim-0.1.6-py2.py3-none-any.whl (4.0 kB)
  Using cached click-8.1.3-py3-none-any.whl (96 kB)
  Using cached future-0.18.2-py3-none-any.whl
  Using cached requests-2.28.1-py3-none-any.whl (62 kB)
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
  Using cached decorator-5.1.1-py3-none-any.whl (9.1 kB)
  Using cached urllib3-1.26.9-py2.py3-none-any.whl (138 kB)
  Using cached idna-3.3-py3-none-any.whl (61 kB)
  Using cached certifi-2022.6.15-py3-none-any.whl (160 kB)
  Using cached charset_normalizer-2.1.0-py3-none-any.whl (39 kB)
  Attempting uninstall: urllib3
    Found existing installation: urllib3 1.26.9
    Uninstalling urllib3-1.26.9:
      Suc

In [2]:
api_key = "AvsDitKm3L7ZEDxEjDf7EUWg9Ne4ze-CwKoWVnymlxZxPK3r1XQa2P_w9hZjcTY4"

In [3]:
import geocoder

In [4]:
df = pd.read_csv("AttractionsFinalData.csv")

In [5]:
df["lat"].value_counts(dropna=False)

nil          347
1.2815683     15
1.29503       14
1.2846547     11
1.3045795      8
            ... 
1.4447509      1
1.304811       1
1.298585       1
1.274996       1
1.2877163      1
Name: lat, Length: 1218, dtype: int64

In [6]:
df["location"].value_counts(dropna=False)

NaN                                                                                          347
18 Marina Gardens Drive Gardens By the Bay Gardens by the Bay, Singapore 018953 Singapore      6
1 Esplanade Drive Esplanade - Theatres On the Bay, Singapore 038981 Singapore                  5
18 Marina Gardens Drive Gardens By the Bay, Singapore 018953 Singapore                         5
10 Bayfront Avenue Marina Bay Sands Marina Bay Sands, Singapore 018956 Singapore               4
                                                                                            ... 
155 Middle Road, Singapore 188977 Singapore                                                    1
Blk 43 Malan Road Gillman Barracks, Singapore 109443 Singapore                                 1
30 Raffles Avenue #01-05 Singapore Flyer, Singapore 039803 Singapore                           1
59 Fort Road, Singapore 439105 Singapore                                                       1
50 Chin Swee Road Thong Chai B

In [8]:
# Fix missing lat, lng and location for locations that are salvageable
df = pd.read_csv("AttractionsFinalData.csv")

ok_data = 0
fixed_data = 0
bad_data = 0

with tqdm(df.iterrows(), total=len(df)) as pbar:
    for i, row in pbar:
        if row.get("lat") == "nil" or row.get("lon") == "nil" or pd.isna(row.get("location", "NaN")):
            bad_data += 1
            g = geocoder.bing(row.get("name") + ", Singapore", key=api_key)
            response = g.json
            # Avoid the generic Singapore, Singapore address that bing search likes to give
            if response is not None and response["address"] != "Singapore, Singapore":
                if pd.isna(row.get("location")):
                    df.at[i, "location"] = response["address"]
                if row.get("lat") == "nil":
                    df.at[i, "lat"] = response["lat"]
                    df.at[i, "lon"] = response["lng"]
                if row.get("formatted_address") == "nil":
                    df.at[i, "formatted_address"] = response["raw"]["address"]["formattedAddress"]
                fixed_data += 1
        else:
            ok_data += 1
        pbar.set_description(f"ok_data: {ok_data}, fixed_data:{fixed_data}/{bad_data}")
        pbar.update(1)

ok_data: 1569, fixed_data:139/347: 100%|██████████| 1916/1916 [01:52<00:00, 17.00it/s]


In [22]:
# Use Google Places API
API_KEY = "AIzaSyAeHanRWhE2iIo-BVvCIy4lEhh1B-bEczY"
base_url = "https://maps.googleapis.com/maps/api/geocode/json?address={}&key={}"

ok_data = 0
fixed_data = 0
bad_data = 0

with tqdm(df.iterrows(), total=len(df)) as pbar:
    for i, row in pbar:
        if row.get("lat") == "nil" or row.get("lon") == "nil" or pd.isna(row.get("location", "NaN")):
            print(row.get("name"))
            bad_data += 1
            url = base_url.format(row.get("name") + "+Singapore", API_KEY)
            response = requests.request("GET", url, headers={}, data={})
            if response is not None and response.status_code == 200:
                response = response.json()
                if len(response["results"]) > 0:
                    if pd.isna(row.get("location")):
                        df.at[i, "location"] = response["results"][0]["address_components"][0]["long_name"]
                    if row.get("lat") == "nil":
                        df.at[i, "lat"] = response["results"][0]["geometry"]["location"]["lat"]
                        df.at[i, "lon"] = response["results"][0]["geometry"]["location"]["lng"]
                    if row.get("formatted_address") == "nil":
                        df.at[i, "formatted_address"] = response["results"][0]["formatted_address"]
                    fixed_data += 1
        else:
            ok_data += 1
        pbar.set_description(f"ok_data: {ok_data}, fixed_data:{fixed_data}/{bad_data}")
        pbar.update(1)


ok_data: 109, fixed_data:1/1:  11%|█         | 212/1916 [00:00<00:02, 731.25it/s] 

Bukit Brown Cemetery
Emirates Lounge


ok_data: 123, fixed_data:3/3:  12%|█▏        | 228/1916 [00:00<00:02, 829.73it/s]

Orchid Garden &amp; Koi Pond
SATS Premier Lounge


ok_data: 169, fixed_data:5/5:  14%|█▍        | 276/1916 [00:00<00:01, 829.73it/s]

Dnata Lounge
Enchanted Garden


ok_data: 207, fixed_data:8/8:  21%|██        | 401/1916 [00:00<00:03, 487.19it/s]

Ambassador Transit Lounge
KrisFlyer Gold Lounge
Central Business District


ok_data: 210, fixed_data:10/10:  21%|██        | 406/1916 [00:01<00:03, 487.19it/s]

Hill Street Building
Marine Cove Playground


ok_data: 228, fixed_data:12/12:  22%|██▏       | 426/1916 [00:01<00:03, 487.19it/s]

Chestnut Nature Park
Cactus Garden


ok_data: 239, fixed_data:14/14:  23%|██▎       | 439/1916 [00:01<00:07, 207.83it/s]

Katong Antique House
New Creation Church


ok_data: 248, fixed_data:16/16:  23%|██▎       | 450/1916 [00:01<00:07, 207.83it/s]

Temple of 1,000 Lights (Sakya Muni Buddha Gaya)
Tekka Temporary Market


ok_data: 280, fixed_data:18/19:  25%|██▌       | 485/1916 [00:02<00:08, 163.74it/s]

City Hall Building
Woodlands Waterfront Park
Hindhede Quarry


ok_data: 298, fixed_data:21/22:  26%|██▋       | 507/1916 [00:02<00:09, 147.32it/s]

Cathay Pacific Airways Lounge
SATS Premier Lounge
SBS Transit - Rail Transit


ok_data: 333, fixed_data:24/25:  28%|██▊       | 544/1916 [00:02<00:09, 144.07it/s]

Chinatown Mid-Autumn Festival Light-Up and Celebrations
Fort Canning Arts Centre


ok_data: 344, fixed_data:25/26:  29%|██▉       | 556/1916 [00:02<00:10, 131.88it/s]

Thai Royal Silk Lounge
Singapore River Festival


ok_data: 357, fixed_data:28/29:  30%|██▉       | 572/1916 [00:03<00:11, 119.10it/s]

Marina Bay Waterfront Promenade
Fragrance Bak Kwa
SATS Premier Lounge


ok_data: 387, fixed_data:31/32:  32%|███▏      | 619/1916 [00:03<00:10, 127.38it/s]

Majestic Fast Ferry
ZhongShan Park
Bedok Jetty


ok_data: 403, fixed_data:34/35:  33%|███▎      | 638/1916 [00:03<00:12, 104.28it/s]

Bird
Standard Chartered Marathon Singapore
Deepavali Celebrations


ok_data: 428, fixed_data:36/37:  35%|███▍      | 665/1916 [00:03<00:13, 95.03it/s] 

Singapore Island Cruise &amp; Ferry Services
Thomson Nature Park


ok_data: 438, fixed_data:39/40:  35%|███▌      | 678/1916 [00:04<00:10, 112.87it/s]

Pandora
Gucci
The First Generation


ok_data: 476, fixed_data:41/43:  38%|███▊      | 719/1916 [00:04<00:09, 131.88it/s]

Hari Raya Light Up
Cocoa &amp; Co
Jelutong Tower


ok_data: 495, fixed_data:44/47:  39%|███▊      | 742/1916 [00:04<00:10, 113.25it/s]

Risis
DBS Asia Treasures Lounge
The River Merchants
Harrods


ok_data: 517, fixed_data:47/50:  40%|████      | 767/1916 [00:04<00:11, 102.05it/s]

PetalClouds
Adidas
DBS Asia Treasures Lounge


ok_data: 549, fixed_data:49/52:  42%|████▏     | 811/1916 [00:05<00:12, 86.66it/s] 

Old Bukit Timah Railway Station
Old Malay and Muslim Cemeteries


ok_data: 568, fixed_data:52/55:  43%|████▎     | 833/1916 [00:05<00:08, 120.48it/s]

Longchamp
Kebun Baru Birdsinging Club
Tua Peh Kong Temple


ok_data: 582, fixed_data:54/58:  44%|████▍     | 850/1916 [00:05<00:09, 109.52it/s]

Rainforest by Sats
Coach
Heritage Zone


ok_data: 601, fixed_data:57/62:  46%|████▌     | 873/1916 [00:05<00:11, 91.64it/s] 

Raffles Garden
Woodlands Temporary Bus Interchange
Burberry
Michael Kors


ok_data: 610, fixed_data:60/65:  46%|████▌     | 886/1916 [00:06<00:12, 79.85it/s]

Adidas Originals
Discover Singapore
DFS Wines &amp; Spirits


ok_data: 615, fixed_data:62/68:  47%|████▋     | 893/1916 [00:06<00:12, 79.85it/s]

Changi Murals
Discover Singapore
Star Island


ok_data: 617, fixed_data:64/70:  47%|████▋     | 897/1916 [00:06<00:16, 62.81it/s]

Night life in krabi
Church of St Ignatius


ok_data: 624, fixed_data:67/73:  47%|████▋     | 907/1916 [00:06<00:20, 49.21it/s]

Mont Blanc
Giordano
iStudio


ok_data: 633, fixed_data:70/76:  48%|████▊     | 919/1916 [00:07<00:19, 51.37it/s]

Maplewood Park
The Pamela Hat sculpture
Stamford House


ok_data: 637, fixed_data:72/79:  48%|████▊     | 926/1916 [00:07<00:24, 39.66it/s]

Watson's Personal Care Store
Kiehl's
Statue: A great emporium, Singapore


ok_data: 660, fixed_data:74/82:  50%|████▉     | 952/1916 [00:07<00:16, 56.99it/s]

Statue: From Chettiars to Financiers, Singapore
DFS Wines &amp; Spirits
District Race


ok_data: 672, fixed_data:77/85:  50%|█████     | 967/1916 [00:07<00:15, 60.64it/s]

Lacoste
Seng Poh Garden &amp; Dancing Girl Sculpture
Memorial to the Victims of Konfrontasi


ok_data: 678, fixed_data:79/87:  51%|█████     | 975/1916 [00:08<00:16, 58.34it/s]

Laichun Yuen Theatre
Fo Shan Ting Da Bo Gong Temple


ok_data: 682, fixed_data:82/90:  51%|█████▏    | 982/1916 [00:08<00:18, 50.26it/s]

Toa Payoh Mall
Bukit Batok Memorial
Fossil


ok_data: 685, fixed_data:84/94:  52%|█████▏    | 989/1916 [00:08<00:22, 40.88it/s]

Joseph Conrad Korzeniowski 顕彰碑
Anglers World Family Fun Park
Gothic Gate
Lee Hwa Diamond Boutique


ok_data: 694, fixed_data:87/97:  52%|█████▏    | 1005/1916 [00:08<00:24, 37.71it/s]

Swarovski
Singapore Art Week
Times Travel


ok_data: 696, fixed_data:90/100:  53%|█████▎    | 1010/1916 [00:09<00:21, 42.54it/s]

The Boy and his Dog
Time Capsule
Farquhar Garden


ok_data: 707, fixed_data:93/103:  53%|█████▎    | 1024/1916 [00:09<00:18, 47.78it/s]

So Chocolate
Bidadari Memorial Garden
Cyclist Park


ok_data: 710, fixed_data:96/106:  54%|█████▍    | 1030/1916 [00:09<00:22, 40.16it/s]

DFS Vintage
Visual Art Trails - Civic District
Marhaba Lounge


ok_data: 719, fixed_data:98/109:  54%|█████▍    | 1042/1916 [00:09<00:20, 43.52it/s]

Inscription of the Island
Relay Newstand
Kidztime!


ok_data: 730, fixed_data:99/112:  55%|█████▌    | 1056/1916 [00:10<00:19, 44.84it/s]

Cosmetics &amp; Perfumes by Shilla
Tiffany &amp; Co.
DFS Watches


ok_data: 759, fixed_data:102/115:  57%|█████▋    | 1088/1916 [00:10<00:18, 45.27it/s]

Mas Golden Lounge
LeSportsac
Avenue Link Transport Group


ok_data: 789, fixed_data:104/117:  58%|█████▊    | 1120/1916 [00:10<00:06, 117.12it/s]

Spectra - A Light And Water Show
Maxi-Cab Singapore


ok_data: 880, fixed_data:107/120:  63%|██████▎   | 1214/1916 [00:10<00:03, 185.15it/s]

The Library
Prime Aces Limousine
Aeroline Service Centre Singapore


ok_data: 890, fixed_data:110/123:  65%|██████▌   | 1246/1916 [00:11<00:03, 196.56it/s]

MyTrip Transport Services
eLimo
Movie Theater


ok_data: 933, fixed_data:112/126:  67%|██████▋   | 1292/1916 [00:11<00:03, 206.75it/s]

MST Transport
Bugis Village
Enjoy Tea


ok_data: 985, fixed_data:114/129:  70%|███████   | 1347/1916 [00:11<00:02, 229.10it/s]

Luxury Coach Services
Cherish Yumiz
Xi Yuan Foot Spa


ok_data: 1003, fixed_data:116/131:  71%|███████▏  | 1367/1916 [00:11<00:02, 188.14it/s]

TJI's Transporter
MrMaxiCab Services


ok_data: 1036, fixed_data:119/134:  73%|███████▎  | 1403/1916 [00:12<00:03, 160.75it/s]

SBS Transit
Ashi Reflexology
Singapore Cab Booking


ok_data: 1145, fixed_data:122/138:  80%|███████▉  | 1531/1916 [00:12<00:01, 252.06it/s]

Cosmetics &amp; Perfumes by Shilla
So Chocolate
My Foot Reflexology
Clementi Woods Park


ok_data: 1161, fixed_data:124/140:  81%|████████  | 1549/1916 [00:12<00:01, 266.36it/s]

Taxibookingonline
Red Dynasty Paintball Park


ok_data: 1207, fixed_data:127/143:  83%|████████▎ | 1598/1916 [00:12<00:01, 178.28it/s]

Discover Singapore
The Rink
Cooking Class Singapore


ok_data: 1220, fixed_data:130/146:  84%|████████▍ | 1614/1916 [00:13<00:01, 178.28it/s]

Boring Indian Food Workshop by madonionslicer (Dhruv Shanker)
SG Hired Cars
Caribbean Spa


ok_data: 1269, fixed_data:133/150:  87%|████████▋ | 1667/1916 [00:14<00:03, 72.50it/s] 

Strides Mobility
Rui Transport Services
Singapore Cab Booking Pte Ltd


ok_data: 1282, fixed_data:134/151:  89%|████████▊ | 1698/1916 [00:14<00:02, 78.68it/s]

Maxi Cab Booking - Limo Taxi
Wessex Estate


ok_data: 1293, fixed_data:137/154:  89%|████████▉ | 1712/1916 [00:14<00:02, 100.27it/s]

Entertainment Deck
Adam Park
M&amp;M's World


ok_data: 1334, fixed_data:139/157:  92%|█████████▏| 1756/1916 [00:15<00:01, 114.30it/s]

Alexandra Park
Seah Im Bunker
LivingveggiebyAnia


ok_data: 1358, fixed_data:141/160:  93%|█████████▎| 1783/1916 [00:15<00:01, 108.43it/s]

Airport Transfer Services in Singapore
Carnival Cinemas
Malayan Leather Craft


ok_data: 1365, fixed_data:145/164:  94%|█████████▎| 1794/1916 [00:15<00:01, 99.54it/s] 

Limousine Services Singapore
Limousine Taxi Singapore
Watson's Personal Care Store
Eng


ok_data: 1427, fixed_data:148/167:  97%|█████████▋| 1859/1916 [00:15<00:00, 134.60it/s]

Shadan Limo Services
Yes Maxi Cab
Kidztime!


ok_data: 1449, fixed_data:150/170:  98%|█████████▊| 1884/1916 [00:16<00:00, 152.52it/s]

Fossil
Linear Park@lentor
Balloon Baron Specialties


ok_data: 1472, fixed_data:153/173: 100%|█████████▉| 1911/1916 [00:16<00:00, 124.91it/s]

Spacio TCM Wellness
Dalvey Estate
Mont Blanc


ok_data: 1499, fixed_data:156/176: : 1940it [00:16, 124.54it/s]                        

Seng Chew Quarry
The Old Jurong Railway Line Trail
Alexandra Canal Linear Park


ok_data: 1521, fixed_data:159/179: : 1977it [00:16, 115.21it/s]

Civic District Tree Trail
Kranji Beach Battle Site
K.STAR Karaoke


ok_data: 1548, fixed_data:161/181: : 2006it [00:17, 149.73it/s]

Sultana Bookstore
Tering Bay Golf and Country Club


ok_data: 1588, fixed_data:165/185: : 2050it [00:17, 104.89it/s]

Quicktransfer
Eternal Spa
Relay Newstand
Blacklane


ok_data: 1612, fixed_data:167/189: : 2078it [00:17, 127.80it/s]

Tiffany &amp; Co.
Movie Theater
Ang Mo Kio Linear Park
Taxi2Airport Singapore


ok_data: 1655, fixed_data:170/192: : 2124it [00:17, 154.47it/s]

Luxe Limo
Singapore-Johore Express
Kaboom


ok_data: 1662, fixed_data:173/195: : 2134it [00:18, 136.98it/s]

Spa Elements
TranSpa
Poolside Bar


ok_data: 1696, fixed_data:176/198: : 2171it [00:18, 106.54it/s]

Book Taxi Singapore
Airport Wellness Oasis
Kallang Bowl


ok_data: 1706, fixed_data:179/201: : 2184it [00:18, 121.10it/s]

Cow Play Cow Moo
iStudio
Star Mart Express


ok_data: 1713, fixed_data:181/203: 100%|██████████| 1916/1916 [00:18<00:00, 101.60it/s]

Mr Lim Foot Reflexology


In [23]:
df["lat"].value_counts(dropna=False)

1.352083     68
nil          22
1.2815683    15
1.29503      14
1.2846547    11
             ..
1.3299929     1
1.309232      1
1.2833627     1
1.2877421     1
1.2877163     1
Name: lat, Length: 1452, dtype: int64

In [45]:
# Store place_id and photo network url using Google Places API
API_KEY = "AIzaSyAeHanRWhE2iIo-BVvCIy4lEhh1B-bEczY"
place_url = "https://maps.googleapis.com/maps/api/place/details/json?place_id={}&fields=place_id%2Cphoto&key={}"
base_url = "https://maps.googleapis.com/maps/api/place/findplacefromtext/json?input={}&inputtype=textquery&fields=place_id%2Cphoto&key={}"

place_ids = []
network_urls = []
df = pd.read_csv("AttractionsFinalData.csv")
with tqdm(df.iterrows()) as pbar:
    for idx, row in pbar:
        name_str = urllib.parse.quote(row.get("name") + " Singapore")
        url = base_url.format(name_str, API_KEY)
        response = requests.request("GET", url, headers={}, data={})
        if response.status_code == 200:
            print(response.text)
            response = response.json()
            place_id = response["candidates"][0]["place_id"]
            place_ids.append(place_id)

0it [00:00, ?it/s]

{
   "candidates" : [
      {
         "photos" : [
            {
               "height" : 3264,
               "html_attributions" : [
                  "\u003ca href=\"https://maps.google.com/maps/contrib/112863337087384359524\"\u003eMinh Nguyen\u003c/a\u003e"
               ],
               "photo_reference" : "Aap_uED7A8dKqrrJ-0hyYoLasVjZeSA5CB5NU6M3Pp7TTiPNjn9nRh-QgVzEgURslRneY0fExjN2Rnc7EvrUdsqd-lxcQ5ctIv-CM8bJmw6bHn85dHo26v9nHo_-VuTEU1TqhXOso73TmyjC0gs02_rC7lYBBiwye0-uyHjrwJdrbRJbLNO8",
               "width" : 4896
            }
         ],
         "place_id" : "ChIJMxZ-kwQZ2jERdsqftXeWCWI"
      }
   ],
   "status" : "OK"
}

{
   "html_attributions" : [],
   "result" : {
      "photos" : [
         {
            "height" : 3264,
            "html_attributions" : [
               "\u003ca href=\"https://maps.google.com/maps/contrib/112863337087384359524\"\u003eMinh Nguyen\u003c/a\u003e"
            ],
            "photo_reference" : "Aap_uEDdBKx0eyGVxkca8IhK0UofH3SfrFtxUS168

In [46]:
photo_reference = "Aap_uED7A8dKqrrJ-0hyYoLasVjZeSA5CB5NU6M3Pp7TTiPNjn9nRh-QgVzEgURslRneY0fExjN2Rnc7EvrUdsqd-lxcQ5ctIv-CM8bJmw6bHn85dHo26v9nHo_-VuTEU1TqhXOso73TmyjC0gs02_rC7lYBBiwye0-uyHjrwJdrbRJbLNO8"
photo_url = "https://maps.googleapis.com/maps/api/place/photo?photo_reference={}&key={}"
url = photo_url.format(photo_reference, API_KEY)

response = requests.request("GET", url, headers={}, data={})
print(response.text)

�PNG

�8D�lF�V��9(¤-Q���IDATx��lU��kb�`1:7$nn�f̈�qι�2I�kv��¨�m2
N��ډ�8�M�@ܘA���{���}{�s�9��\��������<��y���񾷢��K�Q<�(�LHeBʄ�Q&�LHeBʄ��TB�q���D�r�EC�ѣ��au�U��^�o.��Usss��G������	�{�v����s�y���z�$���N=��P8BR�[_�+@�X]D����&��l߾]:t�y��
���=c�XT��y�j?o��ƚ_�W^lҦ^}�^-���2���O��z�)+W,�?��+YB���Z����_A�	��$�����EM�#Gj�"e���w-��]�gE�+*!�UP������"�_���SԄ?g����7���y7[��Dp���6I���ѳ��[�=#�l阤)?�	�8h3N�O<�?!���p�gOm�d]/	�pТ
B���Bڷ�')�F�"�NڌK�큐m�m���O&m3f��a��N���<��TT�Ò��U���Ԭ�YZ`9w�I�As�r
�u��Уg��M=�x_y4�~V,R"�������}�@���@͙9Uuw_�2�ǏI,�)�:h��c����Z_T;n�Qm��(��'z��1%�D۸����zĘ�&sd�L�r5j���,C�go?�3�y%d�…�}�����1���;'x��s�9r�h��͘��J��VTm���)�4>4�0�p�U���:P�5�����ibt����B$�v�"���cm�-%���Ȅ�;!�bn"����V�\x~���5�Uc5���8aFЌ���pi�;mm�'���TK&O��0�ȻS7Ib�̺:u�Ϩ5�?1�1����	"ǎ\��$������j���ci��e�I�Q�e�M����8m�ij�S��~���q��lr�%�����'�/X�����q�X�Hb����G�Br�8d�����R3koӳ(���a�ǰ4e��s�$m\mXX��秄��
����t6`G�
�

In [24]:
df["lon"].value_counts(dropna=False)

103.819836     68
nil            22
103.8636132    15
103.8583026    14
103.8609937    11
               ..
103.843049      1
103.8483525     1
103.8759377     1
103.8578572     1
103.8402478     1
Name: lon, Length: 1445, dtype: int64

In [25]:
df["location"].value_counts(dropna=False)

Singapore                                                                                    92
NaN                                                                                          22
18 Marina Gardens Drive Gardens By the Bay Gardens by the Bay, Singapore 018953 Singapore     6
1 Esplanade Drive Esplanade - Theatres On the Bay, Singapore 038981 Singapore                 5
18 Marina Gardens Drive Gardens By the Bay, Singapore 018953 Singapore                        5
                                                                                             ..
69A Haji Lane, Singapore 189262 Singapore                                                     1
100 Beach Road Shaw Towers, Singapore 189702 Singapore                                        1
Changi                                                                                        1
76 Kim Keat Road Masjid Hajjah Rahimabi Kebun Limau Balestier, Singapore 328835 Singapore     1
50 Chin Swee Road Thong Chai Building, S

In [26]:
df[df["location"] == "Singapore"]

Unnamed: 0.1  Unnamed: 0                                  name  \
6                6           6  Singapore Mass Rapid Transit  (SMRT)   
96              97          97             Ambassador Transit Lounge   
110            111         111                       Emirates Lounge   
199            200         200             Ambassador Transit Lounge   
215            216         216             Central Business District   
...            ...         ...                                   ...   
1859          1862        1862                   Book Taxi Singapore   
1892          1896        1896                Airport Wellness Oasis   
1894          1898        1898                          Kallang Bowl   
1897          1901        1901                      Cow Play Cow Moo   
1899          1903        1903                               iStudio   

       location       lat         lon  \
6     Singapore  1.352083  103.819836   
96    Singapore  1.356428  103.986862   
110   Singapore  1.362107  103.990212   
199   Singapore  1.356428  103.986862   
215   Singapore  1.352083  103.819836   
...         ...       ...         ...   
1859  Singapore  1.352083  103.819836   
1892  Singapore  1.352083  103.819836   
1894  Singapore  1.302192  103.876502   
1897  Singapore  1.295154  103.860095   
1899  Singapore  1.352083  103.819836   

                                      formatted_address  \
6                                             Singapore   
96           Departure Transit Lounge, Singapore 918141   
110                                    Singapore 819643   
199          Departure Transit Lounge, Singapore 918141   
215                                           Singapore   
...                                                 ...   
1859                                          Singapore   
1892                                          Singapore   
1894  5 Stadium Walk #02-22 Leisure Park Kallang, Si...   
1897  3 Temasek Boulevard, Suntec City Tower 3 & 4 #...   
1899                                          Singapore   

                                                  about  overallRating  \
6                                                   NaN            4.5   
96    Tranquility In Transit Experience the warm hos...            3.5   
110                                                 NaN            4.5   
199   Tranquility In Transit Experience the warm hos...            4.0   
215                                                 NaN            4.0   
...                                                 ...            ...   
1859  BookTaxiSingapore is part of Book Taxi Group, ...            2.5   
1892                                                NaN            2.5   
1894                                                NaN            2.5   
1897                                                NaN            2.0   
1899                                                NaN            2.5   

     overallRatingCnt  ...  Water Activities  Alcohol & Night life  Spas  \
6              24,074  ...                 0                     0     0   
96                645  ...                 0                     0     0   
110                60  ...                 0                     0     0   
199               306  ...                 0                     0     0   
215               126  ...                 0                     0     0   
...               ...  ...               ...                   ...   ...   
1859                4  ...                 0                     0     0   
1892               18  ...                 0                     0     1   
1894               18  ...                 0                     0     0   
1897                5  ...                 0                     0     0   
1899               16  ...                 0                     0     0   

      Religion  Food  Shopping  Education  Transport  Fun & Games  Tourism  
6            0     0         0          0          0        

1. Points that are bad are "formatted_address" == "Singapore"
2. Correct points whose "lat" == "1.352083" and "lon" == "103.819836"
3. Points whose "lat" == nil and "lon" == nil

In [32]:
df[df["lon"] == "103.819836"]

Unnamed: 0.1  Unnamed: 0                            name  \
525            526         526                      Ms Redhill   
625            626         626               Water Lily Garden   
1855          1858        1858  EUROPE SHUTTLE Singapore, Asia   

                                               location       lat         lon  \
525   Redhill Wetmarket #01-112, Singapore 150077 Si...  1.352083  103.819836   
625   Terminal 1 Departure Hall, Level 2, Transit Ha...  1.352083  103.819836   
1855            Masti Tn 3-9, Singapore 11911 Singapore  1.352083  103.819836   

     formatted_address                                              about  \
525          Singapore                                                NaN   
625          Singapore  The Water Lily Garden features water lilies in...   
1855         Singapore  EUROPE SHUTTLE offers features to fit in with ...   

      overallRating overallRatingCnt  ...  Water Activities  \
525             5.0                5  ...                 0   
625             4.0                4  ...                 0   
1855            1.0                1  ...                 0   

      Alcohol & Night life  Spas  Religion  Food  Shopping  Education  \
525                      0     0         0     1         0          0   
625                      0     0         0     0         0          0   
1855                     0     0         0     0         0          0   

      Transport  Fun & Games  Tourism  
525           0            0        0  
625           0            0        0  
1855          1            0        0  

[3 rows x 25 columns]

In [26]:
df["formatted_address"].value_counts()

nil                                                               347
3 Temasek Blvd, Singapore 038983                                   15
18 Marina Gardens Dr, Gardens by the Bay, Singapore 018953         15
10 Bayfront Ave, Singapore 018956                                  14
1 HarbourFront Walk, Singapore 098585                               7
                                                                 ... 
315 Outram Rd, #03 02 Tan Boon Liat Building, Singapore 169074      1
90 Hougang Ave 10, Singapore 538766                                 1
Dempsey Rd, Singapore                                               1
525 Balestier Rd, Singapore 329854                                  1
50 Chin Swee Rd, Singapore 169874                                   1
Name: formatted_address, Length: 1244, dtype: int64

In [31]:
df["lat"].value_counts(dropna=False)

1.352083     68
nil          22
1.2815683    15
1.29503      14
1.2846547    11
             ..
1.3299929     1
1.309232      1
1.2833627     1
1.2877421     1
1.2877163     1
Name: lat, Length: 1452, dtype: int64

In [28]:
df["lon"].value_counts(dropna=False)

103.819836     68
nil            22
103.8636132    15
103.8583026    14
103.8609937    11
               ..
103.843049      1
103.8483525     1
103.8759377     1
103.8578572     1
103.8402478     1
Name: lon, Length: 1445, dtype: int64

In [33]:
df.to_csv("AttractionsFinalData.csv")

In [35]:
df["lat"].value_counts(dropna=False)

1.352083     68
nil          22
1.2815683    15
1.29503      14
1.2846547    11
             ..
1.3299929     1
1.309232      1
1.2833627     1
1.2877421     1
1.2877163     1
Name: lat, Length: 1452, dtype: int64

In [36]:
g = geocoder.bing("Bukit Brown Cemetery" + ", Singapore", key=api_key)

In [37]:
print(g.json)

{'address': 'Singapore, Singapore', 'bbox': {'northeast': [1.4763100147247314, 104.09233856201172], 'southwest': [1.1542350053787231, 103.59745025634766]}, 'city': 'Singapore', 'confidence': 1, 'country': 'Singapore', 'lat': 1.28943598, 'lng': 103.84998322, 'ok': True, 'quality': 'PopulatedPlace', 'raw': {'__type': 'Location:http://schemas.microsoft.com/search/local/ws/rest/v1', 'bbox': [1.1542350053787231, 103.59745025634766, 1.4763100147247314, 104.09233856201172], 'name': 'Singapore, Singapore', 'point': {'type': 'Point', 'coordinates': [1.28943598, 103.84998322]}, 'address': {'countryRegion': 'Singapore', 'formattedAddress': 'Singapore, Singapore', 'locality': 'Singapore'}, 'confidence': 'High', 'entityType': 'PopulatedPlace', 'geocodePoints': [{'type': 'Point', 'coordinates': [1.28943598, 103.84998322], 'calculationMethod': 'Rooftop', 'usageTypes': ['Display']}], 'matchCodes': ['Ambiguous', 'UpHierarchy']}, 'status': 'OK'}


In [22]:
no_location = df[pd.isna(df["location"])]

In [38]:
df[df["name"] == "Bukit Brown Cemetery"]

Unnamed: 0.1  Unnamed: 0                  name location       lat  \
102           103         103  Bukit Brown Cemetery      36C  1.336106   

            lon                formatted_address about  overallRating  \
102  103.822985  36C Lor Halwa, Singapore 298637   NaN            4.5   

    overallRatingCnt  ...  Water Activities  Alcohol & Night life  Spas  \
102              230  ...                 0                     0     0   

     Religion  Food  Shopping  Education  Transport  Fun & Games  Tourism  
102         0     0         0          0          0            0        0  

[1 rows x 25 columns]

# Restaurant

In [33]:
food_df = pd.read_csv("RestaurantFinalData.csv")

In [34]:
len(food_df)

7863

In [35]:
food_df["location"].value_counts(dropna=False)

2 Orchard Turn Ion Orchard, Singapore 238801 Singapore                                                              8
9 Raffles Boulevard Millenia Walk, Singapore 039596 Singapore                                                       7
78 Airport Boulevard Jewel Changi Airport, Singapore 819666 Singapore                                               6
10 Scotts Road Grand Hyatt Singapore Grand Hyatt Singapore, Singapore 228211 Singapore                              6
Singapore Singapore                                                                                                 6
                                                                                                                   ..
42 Race Course Road Littile India, Singapore 218557 Singapore                                                       1
14 Lorong Telok, Singapore 049027 Singapore                                                                         1
1 Maritime Square Harbourfront Centre #01-87,#01-88 &amp

In [36]:
food_df["lon"].value_counts(dropna=False)

nil            189
103.8501524     17
103.8332642     16
103.8319512     15
103.8452356     13
              ... 
103.8464431      1
103.9417149      1
103.9039904      1
103.8403343      1
103.8269424      1
Name: lon, Length: 6424, dtype: int64

In [37]:
food_df["lat"].value_counts(dropna=False)

nil          189
1.3037148     17
1.2911354     17
1.3039288     15
1.3005317     13
            ... 
1.302622       1
1.3130158      1
1.3388453      1
1.3139916      1
1.2866719      1
Name: lat, Length: 6443, dtype: int64

In [38]:
food_df

Unnamed: 0.1  Unnamed: 0                       name  \
0                0           0            BAM! Restaurant   
1                1           1        Entre-Nous creperie   
2                2           2              The Courtyard   
3                3           3              Portman's Bar   
4                4           4  Grand Shanghai Restaurant   
...            ...         ...                        ...   
7858         13769       13769            Shi Kou Seafood   
7859         13770       13770            Love, Food Club   
7860         13773       13773       Cafe Bar Tailor Made   
7861         13774       13774                   LiHO TEA   
7862         13775       13775              BLVD @ Changi   

                                               location  \
0     38 Tras Street #38-40, Singapore 078977 Singapore   
1     27 Seah Street # 01-01, Singapore 188383 Singa...   
2     1 Fullerton Square Fullerton Hotel The Fullert...   
3     6 Raffles Boulevard Marina Square Level 4 At P...   
4     390 Havelock Road King's Centre, Singapore 169...   
...                                                 ...   
7858  558 Balestier Road 329874, Singapore 329874 Si...   
7859           9 Perak Road, Singapore 208130 Singapore   
7860  100 Turf Club Rd, Singapore 287992, Singapore ...   
7861  298 Tiong Bahru Road Central Plaza B1-K9, Tion...   
7862  5 Changi Business Park Central 1 Changi City P...   

                                      formatted_address  \
0                          38 Tras St, Singapore 078977   
1                  27 Seah St, #01-01, Singapore 188383   
2                  1 Fullerton Square, Singapore 049178   
3     6 Raffles Boulevard Level 4, PARKROYAL COLLECT...   
4     390 Havelock Rd, Level 1 King's Centre, Singap...   
...                                                 ...   
7858                 558 Balestier Rd, Singapore 329874   
7859                 33 Tessensohn Rd, Singapore 217656   
7860                 100 Turf Club Rd, Singapore 287992   
7861     302 Tiong Bahru Rd, #B1 - k9, Siewiersk 168732   
7862  5 Changi Business Park Central 1, #01-72/73, S...   

                                                   icon  \
0     https://maps.gstatic.com/mapfiles/place_api/ic...   
1     https://maps.gstatic.com/mapfiles/place_api/ic...   
2     https://maps.gstatic.com/mapfiles/place_api/ic...   
3     https://maps.gstatic.com/mapfiles/place_api/ic...   
4     https://maps.gstatic.com/mapfiles/place_api/ic...   
...                                                 ...   
7858  https://maps.gstatic.com/mapfiles/place_api/ic...   
7859  https://maps.gstatic.com/mapfiles/place_api/ic...   
7860  https://maps.gstatic.com/mapfiles/place_api/ic...   
7861  https://maps.gstatic.com/mapfiles/place_api/ic...   
7862  https://maps.gstatic.com/mapfiles/place_api/ic...   

                         place_id price_level        lat          lon  ...  \
0     ChIJSSOENm0Z2jERBKC5c68MTWk         nil   1.278406  103.8442916  ...   
1     ChIJZcsqCqUZ2jERjzIk4OO0wWI           2  1.2962283  103.8545967  ...   
2     ChIJbxecDwkZ2jERUf121CWlCNQ         nil  1.2863437  103.8530812  ...   
3     ChIJt7a1YKgZ2jER64EV7GfGtmY         nil   1.291433  103.8576787  ...   
4     ChIJF9leE50Z2jER4uEdueSxtsM           3  1.2896087  103.8358913  ...   
...                           ...         ...        ...          ...  ...   
7858  ChIJCdy-AGAX2jER0j23uDAFlWo         nil  1.3268054  103.8448565  ...   
7859  ChIJfbMttbkZ2jERO8OM3kNdMRI         nil  1.3159035  103.8571446  ...   
7860  ChIJEyLMId0R2jERuVk87QY06vQ         nil  1.3334509  103.7960173  ...   
7861  ChIJ-66X0X4Z2jERfyZKmfzpFRE         nil  1.2866719  103.8269424  ...   
7862  ChIJ__9zptc82jERXsCfuVnFVQw         nil  1.3335588  103.9633703  ...   

     Fast food Fusion Halal Kosher Pizza Quick Bites Seafood Soups  \
0            0      1     0      0     0           0       0     0   
1            0      0     0      0     0           0       0     0   

In [46]:
imageUrl = []

imageAPIUrlPrefix = "https://serpapi.com/playground?q="
imageAPIUrlSuffix = "&tbm=isch&ijn=0"

with tqdm(food_df.iterrows()) as pbar:
    for i, row in pbar:
        if row.get("lat") == "nil" or row.get("lon") == "nil":
            g = geocoder.bing(row.get("location"), key=api_key)
            response = g.json
            if response is not None and response["address"] != "Singapore, Singapore":
                if row.get("location") == "NaN":
                    food_df.at[i, "location"] = response["address"]
                if row.get("lat") == "nil":
                    food_df.at[i, "lat"] = response["lat"]
                    food_df.at[i, "lon"] = response["lng"]
                if row.get("formatted_address") == "nil":
                    food_df.at[i, "formatted_address"] = response["raw"]["address"]["formattedAddress"]
        response = requests.get(imageAPIUrlPrefix + row.get("name") + imageAPIUrlSuffix)
        if response.status_code == 200:
            print(response.content)
            response = response.json()
            print(response)
        pbar.update(1)

0it [00:02, ?it/s]

b'<!DOCTYPE html>\n<html lang="en">\n  <head>\n    <meta charset="utf-8">\n    <meta name="csrf-param" content="authenticity_token" />\n<meta name="csrf-token" content="a47V5Y76jqhfSxx7Q6DKRcBurXWVR8nOTxb0NMNmJ/s0oEam/69xvkxjwkCoRaS8q9WeZ530QZb7rrXDeaexhw==" />\n\n      \n  <title>SerpApi Playground - SerpApi</title>\n  <meta name="description" content="Test SerpApi&#39;s Google Search, Google Maps, YouTube, Bing, Walmart, Ebay, Baidu, Yandex and more APIs for free in the interactive playground!">\n  <meta name="keywords" content="SerpApi Playground">\n  <meta name="turbolinks-visit-control" content="reload">\n  <script src="/cdn-cgi/apps/head/ICHCiT60g6LBFxQimI2ZxdpAWXU.js"></script><script src="https://www.google.com/recaptcha/api.js?onload=renderRecaptcha&render=explicit" async defer></script>\n\n  <script>\n    function renderRecaptcha() {\n      if (!Cookies.get("api_key")) {\n        RecaptchaLoad();\n      }\n    }\n  </script>\n\n    <meta name="viewport" content="width=device-

JSONDecodeError: Expecting value: line 1 column 1 (char 0)

In [47]:
food_df["lon"].value_counts(dropna=False)

103.8501524    17
103.8332642    16
103.8319512    15
103.8452356    13
103.845069     11
               ..
103.902931      1
103.7700588     1
103.8403289     1
103.840913      1
103.8269424     1
Name: lon, Length: 6591, dtype: int64

In [49]:
food_df["lat"].value_counts(dropna=False)

1.2911354    17
1.3037148    17
1.3039288    15
1.3005317    13
1.3555805    11
             ..
1.353642      1
1.304878      1
1.2843381     1
1.2983644     1
1.2866719     1
Name: lat, Length: 6613, dtype: int64

In [48]:
food_df

Unnamed: 0.1  Unnamed: 0                       name  \
0                0           0            BAM! Restaurant   
1                1           1        Entre-Nous creperie   
2                2           2              The Courtyard   
3                3           3              Portman's Bar   
4                4           4  Grand Shanghai Restaurant   
...            ...         ...                        ...   
7858         13769       13769            Shi Kou Seafood   
7859         13770       13770            Love, Food Club   
7860         13773       13773       Cafe Bar Tailor Made   
7861         13774       13774                   LiHO TEA   
7862         13775       13775              BLVD @ Changi   

                                               location  \
0     38 Tras Street #38-40, Singapore 078977 Singapore   
1     27 Seah Street # 01-01, Singapore 188383 Singa...   
2     1 Fullerton Square Fullerton Hotel The Fullert...   
3     6 Raffles Boulevard Marina Square Level 4 At P...   
4     390 Havelock Road King's Centre, Singapore 169...   
...                                                 ...   
7858  558 Balestier Road 329874, Singapore 329874 Si...   
7859           9 Perak Road, Singapore 208130 Singapore   
7860  100 Turf Club Rd, Singapore 287992, Singapore ...   
7861  298 Tiong Bahru Road Central Plaza B1-K9, Tion...   
7862  5 Changi Business Park Central 1 Changi City P...   

                                      formatted_address  \
0                          38 Tras St, Singapore 078977   
1                  27 Seah St, #01-01, Singapore 188383   
2                  1 Fullerton Square, Singapore 049178   
3     6 Raffles Boulevard Level 4, PARKROYAL COLLECT...   
4     390 Havelock Rd, Level 1 King's Centre, Singap...   
...                                                 ...   
7858                 558 Balestier Rd, Singapore 329874   
7859                 33 Tessensohn Rd, Singapore 217656   
7860                 100 Turf Club Rd, Singapore 287992   
7861     302 Tiong Bahru Rd, #B1 - k9, Siewiersk 168732   
7862  5 Changi Business Park Central 1, #01-72/73, S...   

                                                   icon  \
0     https://maps.gstatic.com/mapfiles/place_api/ic...   
1     https://maps.gstatic.com/mapfiles/place_api/ic...   
2     https://maps.gstatic.com/mapfiles/place_api/ic...   
3     https://maps.gstatic.com/mapfiles/place_api/ic...   
4     https://maps.gstatic.com/mapfiles/place_api/ic...   
...                                                 ...   
7858  https://maps.gstatic.com/mapfiles/place_api/ic...   
7859  https://maps.gstatic.com/mapfiles/place_api/ic...   
7860  https://maps.gstatic.com/mapfiles/place_api/ic...   
7861  https://maps.gstatic.com/mapfiles/place_api/ic...   
7862  https://maps.gstatic.com/mapfiles/place_api/ic...   

                         place_id price_level        lat          lon  ...  \
0     ChIJSSOENm0Z2jERBKC5c68MTWk         nil   1.278406  103.8442916  ...   
1     ChIJZcsqCqUZ2jERjzIk4OO0wWI           2  1.2962283  103.8545967  ...   
2     ChIJbxecDwkZ2jERUf121CWlCNQ         nil  1.2863437  103.8530812  ...   
3     ChIJt7a1YKgZ2jER64EV7GfGtmY         nil   1.291433  103.8576787  ...   
4     ChIJF9leE50Z2jER4uEdueSxtsM           3  1.2896087  103.8358913  ...   
...                           ...         ...        ...          ...  ...   
7858  ChIJCdy-AGAX2jER0j23uDAFlWo         nil  1.3268054  103.8448565  ...   
7859  ChIJfbMttbkZ2jERO8OM3kNdMRI         nil  1.3159035  103.8571446  ...   
7860  ChIJEyLMId0R2jERuVk87QY06vQ         nil  1.3334509  103.7960173  ...   
7861  ChIJ-66X0X4Z2jERfyZKmfzpFRE         nil  1.2866719  103.8269424  ...   
7862  ChIJ__9zptc82jERXsCfuVnFVQw         nil  1.3335588  103.9633703  ...   

     Fast food Fusion Halal Kosher Pizza Quick Bites Seafood Soups  \
0            0      1     0      0     0           0       0     0   
1            0      0     0      0     0           0       0     0   

In [50]:
food_df.to_csv("RestaurantFinalData.csv")

In [2]:
dff = pd.read_csv("Attractions.csv")
print(len(dff))

1828


In [3]:
dff

Unnamed: 0.3  Unnamed: 0.2  Unnamed: 0.1  Unnamed: 0  \
0                0          1669          1672        1672   
1                1           571           572         572   
2                2          1616          1619        1619   
3                3          1656          1659        1659   
4                4          1320          1322        1322   
...            ...           ...           ...         ...   
1823          1911            57            57          57   
1824          1912          1283          1285        1285   
1825          1913           344           345         345   
1826          1914          1633          1636        1636   
1827          1915          1626          1629        1629   

                                 name  \
0                               Zirca   
1     Old Bukit Timah Railway Station   
2                          Go Bambini   
3                   Happy Willow 1 KM   
4                     Le Mandarin Spa   
...                               ...   
1823                     Satay Street   
1824              Clementi Woods Park   
1825                   Marsiling Mall   
1826                   Woodleigh Park   
1827                        N8@yishun   

                                               location       lat         lon  \
0     Clarke Quay Block C The Cannery River Valley R...  1.290025  103.844825   
1                                                     1  1.334269  103.781179   
2     BLK 8 Dempsey Road #01-15 Dempsey Hill, Singap...  1.304311  103.809445   
3     11 Tanjong Katong Road One Km near Paya Lebar ...  1.315047  103.894491   
4     Blk 22 Havelock Road #01-711 Bukit Ho Swee Cou...  1.288885  103.828670   
...                                                 ...       ...         ...   
1823                         Waterloo Street, Singapore  1.299780  103.852390   
1824                                Clementi Woods Park  1.301659  103.766720   
1825        Woodlands Ave 3, Singapore 738623 Singapore  1.432189  103.773704   
1826                          Woodleigh Park, Singapore  1.340090  103.866610   
1827                                  Yishun, Singapore  1.429725  103.835907   

                                      formatted_address  \
0              #01-02 River Valley Rd, Singapore 179020   
1                  #1 Railway Station, Singapore 599938   
2     01-15 Dempsey Rd, BLK 8 Block 8, Singapore 247696   
3     01-22/23 OneKM, 11 Tanjong Katong Rd, Singapor...   
4        01-699 Havelock Rd, BLOCK 22, Singapore 160022   
...                                                 ...   
1823                         Waterloo Street, Singapore   
1824    West Coast Road, Clementi Woods Park, Singapore   
1825                         Woodlands Ave 3, Singapore   
1826                          Woodleigh Park, Singapore   
1827                                  Yishun, Singapore   

                                                  about  ...  \
0     For a sure way to get your senses buzzing, com...  ...   
1                                                   NaN  ...   
2                                                   NaN  ...   
3                                                   NaN  ...   
4     Le Mandarin Spa works to improve quality of li...  ...   
...                                                 ...  ...   
1823                                                NaN  ...   
1824                                                NaN  ...   
1825                                                NaN  ...   
1826                                                NaN  ...   
1827                                                NaN  ...   

      Water Activities Alcohol & Night life  Spas  Religion  Food  Shopping  \
0                    0                    1     0         0     0         0   
1                    0                    0     0         0     0         0   
2                    0                    0     0         0     0         0   
3

## Assign network images

In [ ]:
attraction_df = pd.read_csv("Attractions.csv")
links = []
with open("attraction_links.txt", "r") as f:
    for link in f.readlines():
        links.append(link.strip())
attraction_df["image"] = links

In [ ]:
food_df = pd.read_csv("RestaurantFinalData2.csv")
links = []
with open("restaurant_links.txt", "r") as f:
    for link in f.readlines():
        links.append(link.strip())
food_df["image"] = links